### PR TITLE
MDEV-12459 : Get temporary tables visible to the IS.tables for current connection

### PR DESCRIPTION
**Pull request by an3l**
This PR is related to the MDEV-12459 and all MDEVs that are caused by it (see JIRA).
Most of commits are reviewed, but further review is needed, like MDEV-28334 and MDEV-28453.
Closes PR#695.

### CODE REVIEW

**Comment by an3l** [comment]
@cvicentiu updated to 10.11 and rebased the commits, please review.

**Comment by LinuxJedi** [comment]
@an3l this PR includes too many things in one go and is difficult to review. Can you please break it up to something like one pull request per MDEV?

Also, correct me if I'm wrong, but I believe Monty developed the last patch in the series, but you appear to be marked as the author. If I am correct please fix the author for that commit.

File: sql/sql_show.cc [hunk 1]
```diff
@@ -4800,7 +4805,8 @@ fill_schema_table_by_open(THD *thd, MEM_ROOT *mem_root,
 
 static int fill_schema_table_names(THD *thd, TABLE_LIST *tables,
                                    LEX_CSTRING *db_name,
-                                   LEX_CSTRING *table_name)
+                                   LEX_CSTRING *table_name,
+                                   bool is_temp= false)
```
Comment (cvicentiu) [inline sql/sql_show.cc:4809]: Hi Anel!

I have identified a potential problem with this code.

Can you explain to me what is the behaviour of this function *before* your changes in the cases:

** ASSUME is_sequence is always false **

1a. `ha_table_exists` returns `true`;
2a. `ha_table_exists` returns `false`;

Then, can you explain to me what is the behaviour of this function when:

1b. `ha_table_exists` returns `true`, `is_temp == false`
2b. `ha_table_exists` returns `false`; `is_temp == false`
3b. `ha_table_exists` returns `true`, `is_temp == true`
4b. `ha_table_exists` returns `false`; `is_temp == true`

More specifically I'm interested in knowing what value is stored in `table->field[3]`.

Comment (an3l) [reply sql/sql_show.cc:4809]: Hi @cvicentiu 
`fill_schema_table_names` fills table `IS.table` with data (in your specific question) from base table or temporary table. Before the patch temporary tables were not supported. Wit my patch I used the same function to show temporary tables. Adding `is_temp` allows to handle temporary table that are shadowing the base table (with the same name) and it is up to the the caller to set appropriate flags. We can set `dbug_assert` for situations that may be invoked by the caller (like last combination in below table).
Here is the output you asked:
  | ha_table_exists | is_temp | table->field[3]
-- | -- | -- | --
Before change | 1 | x | Base table
Before change | 0 | x | error
After change | 0 | 0 | error
After change | 0 | 1 | temporary table
After change | 1 | 0 | Base table
After change | 1 | 1 | not allowed/not used

**Comment by an3l** [comment]
Closing this PR, since MDEV is pushed in 11.2

**Pull request closed without merging**